src/test/java/org/cactoos/experimental/ThreadsTest.java

```diff
@@ -92,37 +130,106 @@ void containsResults() {
-            "wraps error into CompletionException",
-            () -> new Threads<String>(
-                Executors.newSingleThreadExecutor(),
-                new ScalarOf<>(
+            "Must contain results from the callables",
+            new Threads<String>(
+                3,
+                Duration.ofSeconds(1),
+                () -> {
+                    this.sleep();
+                    return "txt 1";
+                },
+                () -> {
+                    this.sleep();
+                    return "txt 2";
+                },
+                () -> {
+                    this.sleep();
+                    return "txt 3";
+                }
+            ),
+            new HasValues<>("txt 1", "txt 2", "txt 3")
+        ).affirm();
+    }
+
+    /**
+     * Execution takes longer than timeout when
+     *  {@link ExecutorService} was initiated by {@link Threads} itself.
+     */
+    @RepeatedTest(ThreadsTest.REPETITIONS)
+    void failsDueToTimeoutWithInlineExecutorService() throws Exception {
+        new Assertion<>(
+            "Must fail due to timeout",
+            () -> new LengthOf(
+                new Threads<String>(
+                    2,
+                    Duration.ofMillis(1),
+                    () -> {
+                        this.sleep();
+                        return "txt 1";
+                    },
+                    () -> {
+                        this.sleep();
+                        return "txt 2";
+                    },
                     () -> {
-                        // @checkstyle LineLengthCheck (1 line)
-                        throw new IllegalStateException("Something went wrong");
+                        this.sleep();
+                        return "txt 3";
                     }
                 )
-            ).iterator().next(),
-            new Throws<>(
-                // @checkstyle LineLengthCheck (1 line)
-                "java.io.IOException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Something went wrong",
-                UncheckedIOException.class
-            )
+            ).value(),
+            new Throws<>(CancellationException.class)
         ).affirm();
     }
 
     /**
      * Execute the tasks concurrently using {@link Threads} when
-     *  {@link ExecutorService} was initiated by {@link Threads} itself.
+     *  {@link ExecutorService} was initiated by someone else.
      */
-    @Test
-    void containsValuesWithInlineExecutorService() {
-        this.repeat(
-            arg -> new Assertion<>(
-                // @checkstyle LineLength (1 line)
-                "contains results from the callables when using the inline executor service",
+    @RepeatedTest(ThreadsTest.REPETITIONS)
+    void containsResultsNoTimeout() throws Exception {
+        final ExecutorService extor = Executors.newFixedThreadPool(3);
+        try {
+            new Assertion<>(
+                "Must contain results from callables",
```

Comment:
@victornoel perhaps we could write : "Must contain results from the callables without using timeout" to make a difference with test `containsResults`